distributed/client.py

```diff
@@ -678,6 +671,10 @@ def __init__(
         self._loop_runner = LoopRunner(loop=loop, asynchronous=asynchronous)
         self.loop = self._loop_runner.loop
 
+        self._gather_semaphore = asyncio.Semaphore(5, loop=self.loop.asyncio_loop)
```

Comment:
This object is created by the user outside of the event loop.  The semaphore will grab the event loop of the current thread, which when we're operating in normal synchronous mode, will not be the same event loop as the one that the Client is running on.  So here we have to be explicit and specify the event loop that we want to use.

This usually doesn't come up because we create these objects within async functions or functions that are only being called from within the event loop.  The client code can get strange in this way due to the two threads that are active (user/jupyter/ipython thread and the event loop thread)

Reply:
I see, thanks for the explaination @mrocklin.

Since `Semaphore`, `Condition`, etc. will attach to the current event loop, could temporarily switch the current loop to be the Client's `self.loop.asyncio_loop`, create the `Semaphore`, and then switch the current loop back to what it was before? (Not 100% sure this wouldn't have other consequences, just proposing as a possible option).

Alternatively, I'm happy to keep this as is and open a separate issue for working around the `loop=` keyword depreciation

Reply:
Ah, I missed the statement in your original message that this was being deprecated.  The clean way to do this is to to move this to some `__await__` definition within this class and make sure that we only create the Semaphore in an async function.  I can do this, but I wouldn't mind it being in another PR if that's easy.

Reply:
Gotcha, that does sound cleaner. A separate PR for that is fine